File: LeetCode/[E] - 1137. N-th Tribonacci Number/solution.go
package main

func tribonacci(n int) int {
	dp := make([]int, 3+n)
	dp[0], dp[1], dp[2] = 0, 1, 1
	if n < 3 {
		return dp[n]
	}
	for i := 3; i <= n; i++ {
		dp[i] = dp[i-3] + dp[i-2] + dp[i-1]
	}
	return dp[n]
}

File: LeetCode/[E] - 1137. N-th Tribonacci Number/solution.py
class Solution:
    def tribonacci(self, n: int) -> int:
        dp = [0, 1, 1] + [0] * (n - 2)
        if n < 3:
            return dp[n]
        for i in range(3, n + 1):
            dp[i] = dp[i - 3] + dp[i - 2] + dp[i - 1]
        return dp[n]
